### PR TITLE
Make the example blob renderer multi-threaded.

### PR DESCRIPTION
**Pull request by nical**
Update the blob image renderer example to render all images in parallel using the (infamous) ThreadPool.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1179)

### CODE REVIEW

**Comment by nical** [comment]
cc @jrmuizel r? @kvark

File: webrender/examples/blob.rs [hunk 1]
```diff
@@ -39,32 +44,109 @@ fn deserialize_blob(blob: &[u8]) -> Result<ImageRenderingCommands, ()> {
...
+                    return Err(BlobImageError::Other(format!(
+                        "Usupported image format {:?}",
+                        descriptor.format
+                    )));
+                }
+            }
+        }
+    }
+
+    return Ok(RasterizedBlobImage {
```
Comment (kvark) [inline webrender/examples/blob.rs]: no need for `return`

File: webrender/examples/blob.rs [hunk 2]
```diff
@@ -39,32 +44,109 @@ fn deserialize_blob(blob: &[u8]) -> Result<ImageRenderingCommands, ()> {
...
+        width: descriptor.width,
+        height: descriptor.height,
+    });
+}
+
 struct CheckerboardRenderer {
-    // The deserialized drawing commands.
-    image_cmds: HashMap<ImageKey, ImageRenderingCommands>,
+    // We are going to defer the rendering work to worker threads.
+    // using a pre-built Arc<Mutex<ThreadPool>> rather than creating our own threads
```
Comment (kvark) [inline webrender/examples/blob.rs]: nit: `Using`

File: webrender/examples/blob.rs [hunk 3]
```diff
@@ -76,63 +158,51 @@ impl BlobImageRenderer for CheckerboardRenderer {
...
+                // There it is!
+                return result
+            }
+            self.rendered_images.insert(req, Some(result));
+        }
 
-    fn resolve(&mut self, request: BlobImageRequest) -> BlobImageResult {
-        self.rendered_images.remove(&request).unwrap_or(Err(BlobImageError::InvalidKey))
+        // if we break out of the loop above it means the channel closed unexpectedly.
+        return Err(BlobImageError::Other("Channel closed".into()));
```
Comment (kvark) [inline webrender/examples/blob.rs]: no `return` needed

**Comment by glennw** [comment]
@nical Needs a rebase.

**Comment by nical** [comment]
Yep, I am waiting to see if the switch to rayon sticks before I update this PR for it.

**Comment by glennw** [comment]
@nical We landed the WR update with rayon in servo today - so it should be fine to update this :)

**Comment by glennw** [comment]
Seems to be some compile errors on CI.

**Comment by kvark** [comment]
OSX timeout, retrying
@bors-servo retry

**Pull request merged**